tfe/data_source_oauth_client.go

```diff
@@ -0,0 +1,58 @@
+package tfe
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceTFEOAuthClient() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceTFEOAuthClientRead,
+		Schema: map[string]*schema.Schema{
+			"oauth_client_id": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"ssh_key": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"token_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"api_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"http_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceTFEOAuthClientRead(d *schema.ResourceData, meta interface{}) error {
+	ctx := context.TODO()
```

Comment:
just a question for my own learning: i've never seen this before. i read about it in the docs and it makes sense. is this something we should be doing in all of our resources/data sources that have a blank/empty context?

Reply:
I created an empty context because I needed one to pass to the tfe client.

Perhaps in the long run we could (should?) revisit the codebase and set a context with a timeout (https://godoc.org/context#WithTimeout).